contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -218,19 +195,76 @@ contract BiconomyTokenPaymaster is BasePaymaster, ReentrancyGuard, TokenPaymaste
 
     /**
      * @dev Returns the exchange price of the token in wei.
+     * @param _token ERC20 token address
+     * @param _oracleAggregator oracle aggregator address 
      */
     function exchangePrice(
-        address _token
-    ) public view virtual returns (uint256 exchangeRate) {
-        // get price from oracle aggregator.
+        address _token,
+        address _oracleAggregator
+    ) internal view virtual returns (uint256 exchangeRate) {
+        // get price from chosen oracle aggregator.
         bytes memory _data = abi.encodeWithSelector(IOracleAggregator.getTokenValueOfOneEth.selector, _token);
-        (bool success, bytes memory returndata) = address(oracleAggregator).staticcall(_data);
-        exchangeRate = 0;
+        (bool success, bytes memory returndata) = address(_oracleAggregator).staticcall(_data);
+        exchangeRate = 0; // this is assigned for fallback
         if(success) {
             exchangeRate = abi.decode(returndata, (uint256));
         }
     }
 
+    
+    // review could pack attributes in single / 2 structs. A label field string memory label can be added
+    /**
+     * @dev Helper function to trigger periodic swap to convert tokens received inside this paymaster back to native tokens and immediately deposit on entry point. 
+     * @notice actual swap data is generated offchain basen on i. router ii. token to swap iii. amount of tokens iv. route v. slippage etc
+     * @notice Biconomy may not use feeReceiver to be paymaster contract itself. In this case tokens do not need to be pulled or swapped from the contract
+     * @param _dexRouter dex router/adapter address
+     * @param _swapData calldata for making the swap on chosen router. as function signature also depends on the choice
+     * @param _approveRouter caller sends if the router being used is to be approved or not
+     * @param _token ERC20 token address being swapped. useless is above bool flag is false
+     * @param _amount ERC20 token amount to be approved. useless is above bool flag is false
+     * @param _maxDepositToEP sender can pass maximum amount of gas to be deposited to entry point 
+     */
+    function swapTokenForETHAndDeposit(address _dexRouter, bytes calldata _swapData, bool _approveRouter, address _token, uint256 _amount, uint256 _maxDepositToEP) public nonReentrant onlyOwner returns(bool success, uint256 depositAmount) {
+        // only proceed if adapter is not 0 address
+        require(_dexRouter != address(0), "Router can not be zero address");
+        // make approval to router 
+        if(_approveRouter) {
+        SafeTransferLib.safeApprove(_token, _dexRouter, _amount);
```

Comment:
Formatting issue.

Reply:
prettier plugin will fix it in next push.